packages/api/controllers/user.js

```diff
@@ -20,4 +20,34 @@ const getUserById = async (req, res) => {
   }
 };
 
-module.exports = { getUserById };
+const create = async (req, res) => {
+  const { email, username } = req.body;
+  if (!email) {
+    return res.status(400).send('Missing email');
+  }
+  if (!username) {
+    return res.status(400).send('Missing username');
+  }
+
+  try {
+    const [user, created] = await db.User.findOrCreate({
+      where: {
+        email,
+      },
+      defaults: {
+        username, email,
+      },
+    });
+    if (user) {
+      if (created) return res.status(201).json({ user });
+      return res.status(400).send('User already exists!');
```

Comment:
I'd probably use [409](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409) here.

Reply:
I just added a migration for unique emails, same as for unique names for grant cycle, and the error from sequelize returns `ValidationError` so I'll place it as a 400 type for now. would this be ok since at least it matches the handling of grant cycle?